tracer/src/Datadog.Trace/Activity/ActivityListener.cs

```diff
@@ -0,0 +1,295 @@
+                var activityCurrentDynMethodIl = activityCurrentDynMethod.GetILGenerator();
+                activityCurrentDynMethodIl.EmitCall(OpCodes.Call, activityCurrentMethodInfo, null);
+                activityCurrentDynMethodIl.Emit(OpCodes.Ret);
+                _getCurrentActivity = (Func<object>)activityCurrentDynMethod.CreateDelegate(typeof(Func<object>));
+            }
+
+            static void ChangeActivityDefaultFormat(Type activityType)
+            {
+                // We change the default ID format to W3C (so traceid and spanid gets populated)
+                if (Activator.CreateInstance(activityType, string.Empty).TryDuckCast<IActivityFormat>(out var activityFormat))
+                {
+                    activityFormat.DefaultIdFormat = ActivityIdFormat.W3C;
+                }
+            }
+        }
+
+        public static void StopListeners()
+        {
+            // If there's an activity listener instance we detach the instance and clear it.
+            if (_activityListenerInstance is { } activityListenerInstance)
+            {
+                var activitySourceType = Type.GetType("System.Diagnostics.ActivitySource, System.Diagnostics.DiagnosticSource", throwOnError: true);
+                var detachListenerMethodInfo = activitySourceType!.GetMethod("DetachListener", BindingFlags.Static | BindingFlags.Public);
+                if (detachListenerMethodInfo is not null)
+                {
+                    detachListenerMethodInfo.Invoke(null, new[] { activityListenerInstance });
+                    _activityListenerInstance = null;
+                    Interlocked.Exchange(ref _stopped, 1);
+                }
+                else
+                {
+                    Log.Error("ActivitySource.DetachListener method cannot be found.");
+                }
+            }
+        }
+
+        private static void CreateActivityListenerInstance(Type activityType)
+        {
+            var activityListenerType = Type.GetType("System.Diagnostics.ActivityListener, System.Diagnostics.DiagnosticSource", throwOnError: true)!;
+            var activitySourceType = Type.GetType("System.Diagnostics.ActivitySource, System.Diagnostics.DiagnosticSource", throwOnError: true)!;
+            var activityCreationOptionsType = Type.GetType("System.Diagnostics.ActivityCreationOptions`1, System.Diagnostics.DiagnosticSource", throwOnError: true)!;
+            var activitySamplingResultType = Type.GetType("System.Diagnostics.ActivitySamplingResult, System.Diagnostics.DiagnosticSource", throwOnError: true)!;
+            var activityContextType = Type.GetType("System.Diagnostics.ActivityContext, System.Diagnostics.DiagnosticSource", throwOnError: true)!;
+            var sampleActivityType = Type.GetType("System.Diagnostics.SampleActivity`1, System.Diagnostics.DiagnosticSource", throwOnError: true)!;
```

Comment:
😂 Just disable the nullable if you're going to do this 😛 But I would null propagate everywhere instead and bail out if we need to 😉

Reply:
I mean, I think is impossible that GetType function returns null when using throwOnError right?